arbnode/node.go

```diff
@@ -739,6 +756,51 @@ func createNodeImpl(
 	}, nil
 }
 
+func FindBlockContainingBatchCount(ctx context.Context, bridgeAddress common.Address, l1Client arbutil.L1Interface, parentChainAssertionBlock uint64, batchCount uint64) (uint64, error) {
```

Comment:
I think we need to get Both batchCount and DelayedMessageNumber and use the lowest block that's before both